service/security/test/src/test/resources/features/GroupServiceUnitTests.feature

```diff
@@ -0,0 +1,387 @@
+  Create an Access Group with valid name. Edit the group name to empty name.
+  Kapua should return an error.
+    Given I create the group with name "NewAccessGroup" and description ""
+    Given I expect the exception "KapuaIllegalNullArgumentException" with the text "*"
+    When I update the group name to " "
+    Then An exception was thrown
+
+  Scenario: Editing Access Group name to name that contains numbers
+  Create an Access Group with valid name. Edit the group name to contain numbers. Try to search for it - it should have been changed.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description ""
+    When I update the group name to "Group123"
+    And I search for the group with name "Group123"
+    Then I find the group with name "Group123"
+    And The group was correctly updated
+    And No exception was thrown
+
+  Scenario: Editing Access Group name to name that contains only numbers
+  Create an Access Group with valid name. Edit the group name to contain only numbers. Try to search for it - it should have been changed.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description ""
+    When I update the group name to "1234"
+    And I search for the group with name "1234"
+    Then I find the group with name "1234"
+    And The group was correctly updated
+    And No exception was thrown
+
+  Scenario: Editing Access Group description to unique one
+  Create an Access Group with unique name and with unique description. Once created, search for that description - it should have been changed.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description "description"
+    When I update the group description to "NewDescription"
+    When I search for the group with description "NewDescription"
+    Then I find the group with description "NewDescription"
+    And The group was correctly updated
+    And No exception was thrown
+
+  Scenario: Editing Access Group description to non-unique one
+  Create two Access Groups with different valid names and different description. Edit description of one group to the same description of the second group.
+  Kapua should not return any errors.
+    Given I create the group with name "AccessGroup" and description "description"
+    And I create the group with name "NewAccessGroup" and description "description2"
+    Then I update the group description to "description"
+    Then No exception was thrown
+
+  Scenario: Editing Access Group description to description with special symbols
+  Create unique Access Group with description. Edit the group description to contain valid special symbols, enter all possible characters.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description "description"
+    When I update the group description to "!description@#$%%"
```

Comment:
This is one of the "special" steps - see comments above. Please change this step so it accepts all of the special symbols as parameter or change step name to: "When I update the group description to contain all special symbols" and then in for loop in the implementation go through all the symbols and check if Kapua returns an exception or not.